### PR TITLE
Ajouter les établissements à l'index Elasticsearch

### PR DESCRIPTION
**Pull request by HAEKADI**
closes https://github.com/etalab/annuaire-entreprises-search-infra/issues/70
closes https://github.com/etalab/annuaire-entreprises-search-infra/issues/81
closes https://github.com/etalab/annuaire-entreprises-search-infra/issues/90
closes https://github.com/etalab/annuaire-entreprises-search-infra/issues/83
closes https://github.com/etalab/annuaire-entreprises-search-infra/issues/79
closes https://github.com/etalab/annuaire-entreprises-search-infra/issues/92

### CODE REVIEW

**Review by XavierJp — APPROVED** [review]

File: elasticsearch/mapping_sirene_index.py [hunk 1]
```diff
@@ -57,6 +57,8 @@
     ],
 )
 
+ELASTIC_SHARDS = 1
```
Comment (geoffreyaldebert) [inline elasticsearch/mapping_sirene_index.py:60]: maybe variabilize this but hey this is not the priority

File: helpers/es_fields.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+def get_elasticsearch_field_name(param_name: str) -> str:
```
Comment (geoffreyaldebert) [inline helpers/es_fields.py:1]: cool

File: helpers/utils.py [hunk 3]
```diff
@@ -17,6 +17,15 @@ def str_to_list(string):
     return li
 
 
+def str_to_bool(string):
```
Comment (geoffreyaldebert) [inline helpers/utils.py:20]: your best achievement I think

File: task_functions.py [hunk 4]
```diff
@@ -1024,28 +717,14 @@ def create_spectacle_table():
...
-    ].str[:9]
-    df_spectacle = df_spectacle[["siren", "est_entrepreneur_spectacle"]]
-    df_spectacle = df_spectacle[df_spectacle["siren"].notna()]
+    df_spectacle = preprocess_spectacle_data(DATA_DIR)
     df_spectacle.to_sql("spectacle", siren_db_conn, if_exists="append", index=False)
+    del df_spectacle
+
     for row in siren_db_cursor.execute("""SELECT COUNT() FROM spectacle"""):
         logging.info(
             f"************ {row} records have been added to the SPECTACLE table!"
```
Comment (geoffreyaldebert) [inline task_functions.py:726]: the logging for all of this kind of operation is not exactly correct. Here you count all records of the table (updated each time we operate the cursor), not just the actual new records added. But hey, this is not the priority

File: task_functions.py [hunk 5]
```diff
@@ -1377,33 +852,249 @@ def fill_elastic_index(**kwargs):
...
+                        s.numero_voie_2 as numero_voie_2,
+                        s.siren as siren,
+                        s.siret as siret,
+                        s.tranche_effectif_salarie as tranche_effectif_salarie,
+                        s.type_voie as type_voie,
+                        s.type_voie_2 as type_voie_2
+                        FROM siret s
+                        WHERE s.siren = ul.siren
+                    )
+                ) as etablissements,
```
Comment (geoffreyaldebert) [inline task_functions.py:980]: Too bad there is no nested fields in this nested field, that's not so challenging in fact. But I think you are now master in sql.

File: elasticsearch/process_unites_legales.py [hunk 6]
```diff
@@ -140,40 +111,33 @@ def process_unites_legales(chunk_unites_legales_sqlite):
...
-        unite_legale_processed["liste_uai"] = str_to_list(
-            unite_legale_processed["liste_uai"]
-        )
-        unite_legale_processed["liste_finess"] = str_to_list(
-            unite_legale_processed["liste_finess"]
-        )
+        # Etablissements
+        unite_legale_processed[
+            "etablissements"
+        ] = format_etablissements_and_complements(
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply elasticsearch/process_unites_legales.py:122]: Yeah you're right

File: data_enrichment.py [hunk 7]
```diff
@@ -242,3 +245,92 @@ def create_list_names_elus(list_elus):
...
+                complements[get_elasticsearch_field_name(field)] = True
+
+    return {
+        "etablissements_processed": etablissements_processed,
+        "complements": complements,
+    }
+
+
+# Siege
+def format_siege(siege):
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply data_enrichment.py:312]: Done!

File: elasticsearch/indexing_unite_legale.py [hunk 8]
```diff
@@ -46,14 +46,14 @@ def index_unites_legales_by_chunk(
         if logger % 1000 == 0:
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply elasticsearch/indexing_unite_legale.py:46]: I increased it to 100.000 🤭

File: elasticsearch/mapping_sirene_index.py (hunk 1 above)
Comment (HAEKADI) [reply elasticsearch/mapping_sirene_index.py:60]: I'll do it in a separate PR. I did not want to overload this one even more than I already did 👉👈

File: elasticsearch/mapping_sirene_index.py [hunk 9]
```diff
@@ -74,6 +76,118 @@ class ElasticsearchDirigeantPMIndex(InnerDoc):
...
+    )
+    coordonnees = GeoPoint()
+    complement_adresse = Text()
+    complement_adresse_2 = Text()
+    date_creation = Date()
+    date_debut_activite = Date()
+    departement = Keyword()
+    distribution_speciale = Text()
+    distribution_speciale_2 = Text()
+    enseigne_1 = Text(analyzer=annuaire_analyzer, fields={"keyword": Keyword()})
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply elasticsearch/mapping_sirene_index.py:101]: 😆 okay I'll add a comment

File: elasticsearch/mapping_sirene_index.py [hunk 10]
```diff
@@ -94,79 +208,47 @@ class ElasticsearchSireneIndex(Document):
...
     nature_juridique_unite_legale = Integer()
     nom = Text(analyzer=annuaire_analyzer)
     nom_complet = Text(analyzer=annuaire_analyzer, fields={"keyword": Keyword()})
     nom_raison_sociale = Text()
     nombre_etablissements = Integer()  # NaN can't be stored in an integer array
     nombre_etablissements_ouverts = Integer()
-    numero_voie = Text()
     prenom = Text(analyzer=annuaire_analyzer)
     section_activite_principale = Keyword()
+    siege = Nested(ElasticsearchSiegeIndex)
```
[the remark this answers is not in the document]
Comment (HAEKADI) [reply elasticsearch/mapping_sirene_index.py:241]: Don't I know it 🤦‍♀️

File: helpers/es_fields.py (hunk 2 above)
Comment (HAEKADI) [reply helpers/es_fields.py:1]: Cool is my middle name 🤓

File: helpers/utils.py (hunk 3 above)
Comment (HAEKADI) [reply helpers/utils.py:20]: hahaha `bool`and `SQLight` in the same sentence is my worst nightmare second only to `nested` and `filter`....

File: task_functions.py (hunk 4 above)
Comment (HAEKADI) [reply task_functions.py:726]: Stop being so passive aggressive about allll the things I still have to do 🤣

File: task_functions.py (hunk 5 above)
Comment (HAEKADI) [reply task_functions.py:980]: image

**Pull request merged**